controllers/user.js

```diff
@@ -4,50 +4,98 @@ const randomToken = require('random-token')
-                    .then(() => {
-                        res.status(200).send(`User: "${req.body.name}" - registered :)`);
-                    })
-            })
-        } else {
-            return res.status(400).send(`User name "${req.body.name}" already exist :(`); // 400 == Bad request
-        }
-    } catch (err) {
-        console.error(err);
-    }
+exports.signup = (req, res) => {
+    const {
+        name,
+        passw
+    } = req.body;
+    User.findOne({
+            name: name
+        })
+        .then(user => {
+            if (user) {
+                return res.status(400).json({
+                    user: user.name,
+                    message: 'already exist'
+                });
+            }
+            return bCrypt.hash(passw, 10);
+        })
+        .then((hash) => {
+            const user = new User({
+                name,
+                passw: hash
+            });
+            return user.save();
+        })
+        .then((user) => res.status(200).json({
+            user: user.name,
+            message: 'registered'
+        }))
+        .catch((err) => console.error(err))
 };
 
 
 exports.login = (req, res) => {
-    try {
-        User.findOne({ name: req.body.name })
-            .then((user) => {
-                if (user) {
-                    bCrypt.compare(req.body.passw, user.passw, (err, result) => {
-                        if (result) {
-                            //token ?
-                            res.status(200).send('User logged in ;)');
-                        } else {
-                            res.status(404).send('Invalid credentials :(');
-                        }
-                    });
-                } else {
-                    res.status(400).send('Invalid credentials :(')
-                }
+    const {
+        name,
+        passw
+    } = req.body;
+    User.findOne({
+            name: name
+        })
+        .then(user => {
+            if (!user) {
+                return res.status(400).json({
+                    message: 'Invalid credentials'
+                });
+            }
+            return bCrypt.compare(passw, user.passw);
+        })
+        .then((result) => {
+            if (!result) {
+                return res.status(400).json({
+                    message: 'Invalid credentials'
+                });
+            }
+            const token = randomToken(24);
+            return User.findOneAndUpdate({
+                name
+            }, {
+                token: token
+            }, {
+                new: true
             });
-    }
-    catch (err) {
-        console.error(err);
-    }
+        })
+        .then((user) => res.status(200).json({
+            message: 'User logged in',
+            token: user.token
+        }))
+        .catch((err) => console.error(err))
 };
 
 
 exports.logout = (req, res) => {
-    return res.status(200).send('User logged out :)');
+    const authHeader = req.get('Authorization');
```

Comment:
Now we send token in token field in the header. Change req.get('Authorization') to req.headers.token

Reply:
changed.